WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -262,6 +262,32 @@ extension OrderDetailsViewModel {
             self.syncSubscriptions { _ in
                 group.leave()
             }
+
+            // Shipping labels need to be synced after the order but before we complete
+            // the order sync group to ensure the UI shows the latest data
+            group.enter()
+            Task { @MainActor [weak self] in
+                defer {
+                    group.leave()
+                }
+                // Check Woo Shipping support first, to ensure correct flows are enabled for shipping labels.
+                self?.dataSource.isEligibleForWooShipping = ((await self?.isWooShippingSupported()) != nil)
```

Comment:
We could unwrap `self` using a guard statement to avoid using self as an optional here and a few other places below.

Reply:
Done in bbfb18c0b00fd7ba7a54318eb920f8ec66e13d92.